### PR TITLE
Feature/aql subquery execution block impl execute implementation shortest path executor

### PR DESCRIPTION
**Pull request by mchacki**
Internal PR.
Original author is @markuspf 
I am just the reviewer.

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins:
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/7863/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8126/

File: tests/Aql/ShortestPathExecutorTest.cpp [hunk 1]
```diff
@@ -175,762 +185,265 @@ struct TestShortestPathOptions : public ShortestPathOptions {
...
-}
+    ValidateResult(output, skipped);
+  }
+};
 
-TEST_F(ShortestPathExecutorTest,
-       NotWaiting_WithMultipleRows_UsingVertexAndEdgeOutput_UsingRegisterSourceInput_UsingRegisterTargetInput) {
-  RunTestWithMultipleRowsUpstream(false, std::move(regSource), std::move(regTarget), true);
-}
+TEST_P(ShortestPathExecutorTest, the_test) { TestExecutor(infos, input); }
```
Comment (mchacki) [inline tests/Aql/ShortestPathExecutorTest.cpp]: Can you please add a comment how this test works, what does it test and why is it enough?
If you read it long enough it gets clear how it works, but that is rather obfuscated.

Also please add (a/some) test(s) for fullCount.
With fullCount you will get into skip/get/fullCount situation, where I think a bug is hidden.

Also can you add tests for multiple counts?

produce => outputFull, halfPathExists
produce(again) => halfPath continued.

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8259/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8272/

**Pull request merged**